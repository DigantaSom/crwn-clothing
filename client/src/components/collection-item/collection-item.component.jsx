import { connect } from 'react-redux';
import { addItem } from '../../redux/cart/cart.actions';

import CustomButton from '../custom-button/custom-button.component';

import './collection-item.styles.scss';

export const CollectionItem = ({ item, addItem }) => {
  const { imageUrl, name, price } = item;

  return (
    <div className='collection-item'>
      <div className='image' style={{ backgroundImage: `url(${imageUrl})` }} />
      <div className='collection-footer'>
        <div className='name'>{name}</div>
        <div className='price'>{price}</div>
      </div>
      <CustomButton id='add-button' onClick={() => addItem(item)} inverted>
        Add to Cart
      </CustomButton>
    </div>
  );
};

const mapDispatchToProps = dispatch => ({
  addItem: item => dispatch(addItem(item)),
});

export default connect(null, mapDispatchToProps)(CollectionItem);
