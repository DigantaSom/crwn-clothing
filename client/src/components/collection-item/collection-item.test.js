import { shallow } from 'enzyme';

import { CollectionItem } from './collection-item.component';

describe('CollectionItem component', () => {
  let wrapper;
  let mockAddItem;
  const mockImageUrl = 'www.testImage.com';
  const mockName = 'black hat';
  const mockPrice = 10;

  beforeEach(() => {
    mockAddItem = jest.fn();

    const mockProps = {
      item: {
        imageUrl: mockImageUrl,
        name: mockName,
        price: mockPrice,
      },
      addItem: mockAddItem,
    };

    wrapper = shallow(<CollectionItem {...mockProps} />);
  });

  it('should render CollectionItem component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call addItem when add-to-cart-button is clicked', () => {
    wrapper.find('#add-button').simulate('click');
    expect(mockAddItem).toHaveBeenCalled();
  });

  // The below test works when styled-components are used for this component.
  // it('should render imageUrl as a prop on BackgroundImage', () => {
  // 	expect(wrapper.find('BackgroundImage').prop('imageUrl')).toBe(mockImageUrl);
  // })

  it('should render name in div.name', () => {
    expect(wrapper.find('.name').text()).toBe(mockName);
  });

  it('should render price in div.price', () => {
    const price = parseInt(wrapper.find('.price').text());
    expect(price).toBe(mockPrice);
  });
});
