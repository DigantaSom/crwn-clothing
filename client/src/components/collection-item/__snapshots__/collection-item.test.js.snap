// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem component should render CollectionItem component 1`] = `
<div
  className="collection-item"
>
  <div
    className="image"
    style={
      Object {
        "backgroundImage": "url(www.testImage.com)",
      }
    }
  />
  <div
    className="collection-footer"
  >
    <div
      className="name"
    >
      black hat
    </div>
    <div
      className="price"
    >
      10
    </div>
  </div>
  <CustomButton
    id="add-button"
    inverted={true}
    onClick={[Function]}
  >
    Add to Cart
  </CustomButton>
</div>
`;
